kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -485,14 +485,9 @@ impl<'tcx> GotocCtx<'tcx> {
             Rvalue::Aggregate(ref k, operands) => {
                 self.codegen_rvalue_aggregate(k, operands, res_ty)
             }
-            Rvalue::ThreadLocalRef(_) => {
-                let typ = self.codegen_ty(res_ty);
-                self.codegen_unimplemented(
-                    "Rvalue::ThreadLocalRef",
-                    typ,
-                    Location::none(),
-                    "https://github.com/model-checking/kani/issues/541",
-                )
+            Rvalue::ThreadLocalRef(def_id) => {
+                // Since Kani is single-threaded, we treat a thread local like a static variable:
+                self.codegen_static_pointer(*def_id)
```

Comment:
Should probably set `    pub is_thread_local: bool,` on the struct
https://github.com/model-checking/kani/blob/main/cprover_bindings/src/goto_program/symbol.rs#L42

Reply:
Good catch. Done.